packages/flutter/test/material/dropdown_menu_test.dart

```diff
@@ -3901,6 +3901,29 @@ void main() {
     },
     variant: TargetPlatformVariant.all(),
   );
+
+  testWidgets('DropdownMenu passes maxLines to TextField', (WidgetTester tester) async {
+    Widget buildDropdownMenu({int? maxLines = 1}) {
+      return MaterialApp(
+        home: Scaffold(
+          body: DropdownMenu<TestMenu>(dropdownMenuEntries: menuChildren, maxLines: maxLines),
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildDropdownMenu());
+    TextField textField = tester.widget(find.byType(TextField));
+    // Default behavior.
+    expect(textField.maxLines, 1);
+
+    await tester.pumpWidget(buildDropdownMenu(maxLines: null));
+    textField = tester.widget(find.byType(TextField));
+    expect(textField.maxLines, null);
+
+    await tester.pumpWidget(buildDropdownMenu(maxLines: 2));
```

Comment:
```suggestion
await tester.pumpWidget(
      MaterialApp(home: Scaffold(body: DropdownMenu<TestMenu>(dropdownMenuEntries: menuChildren))),
    );
    TextField textField = tester.widget(find.byType(TextField));
    // Default behavior.
    expect(textField.maxLines, 1);

    await tester.pumpWidget(
      MaterialApp(
        home: Scaffold(
          body: DropdownMenu<TestMenu>(dropdownMenuEntries: menuChildren, maxLines: null),
        ),
      ),
    );
    textField = tester.widget(find.byType(TextField));
    expect(textField.maxLines, null);

    await tester.pumpWidget(
      MaterialApp(
        home: Scaffold(
          body: DropdownMenu<TestMenu>(dropdownMenuEntries: menuChildren, maxLines: 2),
        ),
      ),
    );
```

Default behavior check is actually passing a value of 1 here so I refactored this to pump widget code multiple times since the widget code is pretty short in this test.

Reply:
The functional version looks much cleaner.

Reply:
With functions, default value isn't tested, it is always passed value.  Feel free to add check without passing a value then keep function for remaining checks. I just refactored to exclude function the widget code is just a couple of lines.